test/PriceConsumerV3_test.js

```diff
@@ -4,20 +4,31 @@ const { networkConfig } = require('../helper-hardhat-config')
 
 describe('PriceConsumer', async function () {
   // Price Feed Address, values can be obtained at https://docs.chain.link/docs/reference-contracts
-  let priceConsumerV3, ethUsdAggregator
+  let priceConsumerV3, ethUsdAggregator, PriceConsumerV3, MockV3Aggregator
 
   beforeEach(async () => {
-    await deployments.fixture(['mocks', 'feed'])
+    // We ONLY deploy the fixtures on a local test
+    // Also, Aggregator contract is only needed for local test
+    if (await getChainId() == 31337) {
+      await deployments.fixture(['mocks', 'feed'])
+      EthUsdAggregator = await deployments.get("EthUsdAggregator")
+      ethUsdAggregator = await ethers.getContractAt("MockV3Aggregator", EthUsdAggregator.address)
+    }
 
     // Then, we can get the contracts that were just deployed
-    const PriceConsumerV3 = await deployments.get('PriceConsumerV3')
-    priceConsumerV3 = await ethers.getContractAt('PriceConsumerV3', PriceConsumerV3.address)
-    const EthUsdAggregator = await deployments.get('EthUsdAggregator')
-    ethUsdAggregator = await ethers.getContractAt('MockV3Aggregator', EthUsdAggregator.address)
+    const PriceConsumerV3 = await deployments.get("PriceConsumerV3")
+    priceConsumerV3 = await ethers.getContractAt("PriceConsumerV3", PriceConsumerV3.address)
+
   })
 
   it('should return a positive value', async () => {
-    expect(await priceConsumerV3.getLatestPrice())
-      .to.equal((await ethUsdAggregator.latestRoundData())[1])
+    if (await getChainId() == 31337) {
```

Comment:
Let's not have an "if local" here. Let's have the local and tesnet tests try to be the same whenever possible.

Reply:
Price feeds is hard to have the same uniform test for both local and public networks because of the way the deployments are done. Local uses MockAggregator with some static value as the price, then we compare the deployed price feed price to that number, whereas public network uses the already deployed aggregators, and just checks for a  >0 value, which means that it was deployed properly and is looking at the aggregator specified in the constructor (although we don't really know if the number is correct)

The public network test can't compare the price feed value to the aggregator contract like the local one (as far as I'm aware), but the local one can just check for a >0 value as opposed to comparing the price feed latest price to the mock aggregator latestRoundData, so for now I'll go with that, but let me know if you have any other suggestions